src/containers/App/App.test.js

```diff
@@ -1,9 +1,67 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './App';
+import { App, mapDispatchToProps } from './App';
+import API from '../../utils/api';
+import { apiKey } from '../../utils/api-key';
+import { shallow } from 'enzyme';
+import { getMovies } from '../../actions';
 
 it('renders without crashing', () => {
   const div = document.createElement('div');
   ReactDOM.render(<App />, div);
   ReactDOM.unmountComponentAtNode(div);
 });
+
+describe('App', () => {
+  describe('defaults', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      wrapper = shallow(<App />);
+    });
+
+    it('should match the snapshot with all data passed in', () => {
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('componentDidMount', () => {
+    let wrapper;
+    let mockData;
+    let mockGetMovies;
+
+    beforeEach(() => {
+      mockGetMovies = jest.fn();
+      mockData = { results: [{ name: 'AquaMan', id: 1 }, { name: 'Beauty and the Beast', id: 2 }] };
+      API.fetchData = jest.fn().mockImplementation(() => {
+        return mockData
+      });
+      wrapper = shallow(
+      <App 
+        getMovies={mockGetMovies}
+      />);
+    });
+
+    it('should call fetchData with the correct parameters', async () => {
+      const expected = `https://api.themoviedb.org/3/movie/popular?page=1&api_key=${apiKey}&language=en-US`;
+      await wrapper.instance().componentDidMount();
```

Comment:
I like that you can wait for the component to mount, I was not aware of that for testing purposes